### PR TITLE
Add developer docs

### PR DESCRIPTION
**Pull request by rockdaboot**
This PR adds some documentation:
- Update dev branch with changes from upstream
- Upgrade dev branch to next Kibana version
- Build and push the ES cloud docker image
- Build and push the Kibana cloud docker image

### CODE REVIEW

**Comment by rockdaboot** [comment]
It is based on 8.3+profiler because I currently don't want to checkout 8.2* as this involves expensive bootstrap and build operations.
I'll cherry-pick the commit for the 8.2+profiler branch manually later.

File: src/plugins/profiling/README.md [hunk 1]
```diff
@@ -62,3 +62,51 @@ You should expect some of these commands to take several minutes to finish.
...
+Make sure the remote `upstream` exists. If it doesn't, create it with `git remote add upstream https://github.com/elastic/kibana`.
+
+Example on how upgrade from 8.2 to 8.3:
+- get latest changes from all remotes with `git fetch --all --prune`
+- create and checkout the new development branch with `git checkout -b 8.3+profiler upstream/8.3`
+- cherry-pick all our changes from `8.2+profiler` with e.g. `git cherry-pick 10fdea5a7156ca0d576142e78ab0d17dc7420b1d^..8.2+profiler`, where that commit hash refers to our first commit (created by Stephanie at 07 Dec "add initial profiling search").
+- fix the merge conflicts that are likely popping up. Normally, these are in `package.json` and `yarn.lock`.
+- when done, push the new branch to our development repository with `git push --set-upstream origin 8.3+profiler`
+
+### Build and push the ES cloud docker image
```
Comment (inge4pres) [inline src/plugins/profiling/README.md:91]: We should add here the pre-requisite of logging into the registry, linking the infra doc (it's mentioned in the eng@ thread with subject "Environment for MVP"

**Review by inge4pres — COMMENTED** [review]
Do you mind adding also instructions on how to deploy to MVP?
Or should we reserve that for another PR?

File: src/plugins/profiling/README.md (hunk 1 above)
Comment (rockdaboot) [reply src/plugins/profiling/README.md:91]: :+1: Updated

File: src/plugins/profiling/README.md [hunk 2]
```diff
@@ -62,3 +62,53 @@ You should expect some of these commands to take several minutes to finish.
...
+- push the docker image with `docker push docker.elastic.co/observability-ci/elasticsearch-profiling:8.3.0-$(git rev-parse --short HEAD)`
+- for later deployment, take note of the output of `echo docker.elastic.co/observability-ci/elasticsearch-profiling:8.3.0-$(git rev-parse --short HEAD)`
+
+### Build and push the Kibana cloud docker image
+
+In order to deploy a custom Kibana build in our MVP or testing cluster, we have to build the Kibana cloud docker image locally and push it to the Elastic docker registry.
+You'll need a login to the Elastic container library. [Follow this doc to get access to the registry](https://github.com/elastic/infra/blob/master/docs/container-registry/accessing-the-docker-registry.md).
+
+From within the local `kibana` repository:
+- checkout the branch you want ot deploy with e.g. `git checkout 8.3+profiler`
```
Comment (jbcrail) [inline src/plugins/profiling/README.md]:
```suggestion
- checkout the branch you want to deploy with e.g. `git checkout 8.3+profiler`
```

File: src/plugins/profiling/README.md [hunk 3]
```diff
@@ -62,3 +62,53 @@ You should expect some of these commands to take several minutes to finish.
...
+- if you didn't do this before on your branch: `yarn kb reset && yarn kbn bootstrap`
```
Comment (jbcrail) [inline src/plugins/profiling/README.md]:
```suggestion
- if you didn't do this before on your branch: `yarn kbn reset && yarn kbn bootstrap`
```

File: src/plugins/profiling/README.md [hunk 4]
```diff
@@ -62,3 +62,53 @@ You should expect some of these commands to take several minutes to finish.
...
+- fix the merge conflicts that are likely popping up. Normally, these are in `package.json` and `yarn.lock`.
+- when done, push the new branch to our development repository with `git push --set-upstream origin 8.3+profiler`
+
+### Build and push the ES cloud docker image
+
+In order to deploy a custom ES build in our MVP or testing cluster, we have to build the ES cloud docker image locally and push it to the Elastic docker registry.
+You'll need a login to the Elastic container library. [Follow this doc to get access to the registry](https://github.com/elastic/infra/blob/master/docs/container-registry/accessing-the-docker-registry.md).
+
+From within the local `elasticsearch` repository:
+- checkout the branch you want ot deploy with e.g. `git checkout prodfiler`
```
Comment (jbcrail) [inline src/plugins/profiling/README.md]:
```suggestion
- checkout the branch you want to deploy with e.g. `git checkout prodfiler`
```

**Review by jbcrail — CHANGES_REQUESTED** [review]

**Pull request merged**